ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,182 @@
+        },
+      ];
+    case 'pki':
+      return [
+        {
+          actionTitle: 'Issue certificate',
+          actionType: 'issue-certificate-pki',
+        },
+        {
+          actionTitle: 'View certificate',
+          actionType: 'view-certificate-pki',
+        },
+        {
+          actionTitle: 'View issuer',
+          actionType: 'view-issuer-pki',
+        },
+      ];
+  }
+};
+
+const getActionParamByAction = (type) => {
+  switch (type) {
+    case 'find-kv':
+      return {
+        title: 'Secret Path',
+        subText: 'Path of the secret you want to read, including the mount. E.g., secret/data/foo.',
+        elementType: 'input',
+        buttonText: 'Read secrets',
+        path: 'vault.cluster.secrets',
+      };
+    case 'generate-credentials-db':
+      return {
+        title: 'Role to use',
+        elementType: 'search-select',
+        buttonText: 'Generate credentials',
+        model: 'database/role',
+        path: 'vault.cluster.secrets.backend.credentials',
+      };
+    case 'issue-certificate-pki':
+      return {
+        title: 'Role to use',
+        elementType: 'search-select',
+        placeholder: 'Type to find a role...',
+        buttonText: 'Issue leaf certificate',
+        model: 'pki/role',
+        path: 'vault.cluster.secrets.backend.pki.roles.role.generate',
+      };
+    case 'view-certificate-pki':
+      return {
+        title: 'Certificate serial number',
+        placeholder: '33:a3:...',
+        elementType: 'search-select',
+        buttonText: 'View certificate',
+        model: 'pki/certificate/base',
+        path: 'vault.cluster.secrets.backend.pki.certificates.certificate.details',
+      };
+    case 'view-issuer-pki':
+      return {
+        title: 'Issuer',
+        placeholder: 'Type issuer name or ID',
+        elementType: 'search-select',
+        buttonText: 'View issuer',
+        model: 'pki/issuer',
+        nameKey: 'issuerName',
+        path: 'vault.cluster.secrets.backend.pki.issuers.issuer.details',
+      };
+  }
+};
+
+export default class DashboardQuickActionsCard extends Component {
+  @service router;
+
+  @tracked selectedEngine;
+  @tracked selectedActions = [];
+  @tracked selectedAction;
+  @tracked actionParamField;
+  @tracked selectedEngineName;
+  @tracked value;
+
+  constructor() {
+    super(...arguments);
+
+    if (!this.selectedEngine) {
+      this.setSelectedAction();
+    }
+  }
+
+  get filteredSecretEngines() {
+    return this.args.secretsEngines.filter(
+      (secretEngine) =>
+        (secretEngine.shouldIncludeInList && secretEngine.type === 'pki') ||
+        secretEngine.type === 'kv' ||
+        secretEngine.type === 'database'
+    );
+  }
+
+  get secretsEnginesOptions() {
+    return this.filteredSecretEngines.map((filteredSecretEngine) => {
+      const name = filteredSecretEngine.path.replace('/', '');
+
+      return {
+        name,
+        id: `${filteredSecretEngine.type} ${name}`,
```

Comment:
so this would work out to be something like `kv kv-engine`. I can't quite remember how SearchSelect handles ids, but is that correct?

Reply:
yup, it would be and i don't feel like it's ideal for display so i'm planning to have an arg that will hide this 
<img width="532" alt="Screenshot 2023-08-01 at 1 14 08 PM" src="https://github.com/hashicorp/vault/assets/30884335/75ee78e8-7fa5-4152-86cb-2b7d4112322a">

Reply:
I have feedback on this portion before you refactor too much but I'm curious what the intention design-wise is? Typically we show the id, or name + id if those are different, but that's not the case here since the engine name _is_ the id

one option is showing the engine name and engine type: 
<img width="546" alt="Screenshot 2023-08-02 at 5 09 08 PM" src="https://github.com/hashicorp/vault/assets/68122737/821f8fc6-b16c-416e-b15e-7614e7e6279e">